core/src/test/java/org/apache/calcite/test/RelMetadataTest.java

```diff
@@ -2506,6 +2506,30 @@ private void checkPredicates(RelOptCluster cluster, RelOptTable empTable,
     assertNull(r);
   }
 
+  @Test void testExpressionLineageCalc() {
+    final RelNode rel = convertSql("select sal from (\n"
+        + " select * from emp) "
+        + "where deptno = 10");
+    final RelNode tableRel = convertSql("select * from emp");
+    final HepProgramBuilder programBuilder = HepProgram.builder();
+    programBuilder.addRuleInstance(CoreRules.PROJECT_TO_CALC);
+    programBuilder.addRuleInstance(CoreRules.FILTER_TO_CALC);
+    programBuilder.addRuleInstance(CoreRules.CALC_MERGE);
+    final HepPlanner planner = new HepPlanner(programBuilder.build());
+    planner.setRoot(rel);
+    final RelNode optimizedRel = planner.findBestExp();
+    final RelMetadataQuery mq = optimizedRel.getCluster().getMetadataQuery();
+
+    final RexNode ref = RexInputRef.of(0, optimizedRel.getRowType().getFieldList());
+    final Set<RexNode> r = mq.getExpressionLineage(optimizedRel, ref);
+
+    final String inputRef = RexInputRef.of(5, tableRel.getRowType().getFieldList()).toString();
+    assertThat(r.size(), is(1));
+    final String resultString = r.iterator().next().toString();
+    assertThat(resultString, startsWith(EMP_QNAME.toString()));
+    assertThat(resultString, endsWith(inputRef));
```

Comment:
I'm afraid it is obscure what is the expected output and why is that expected. Do you think you can convert the expected value to a single string with a corresponding message that clarifies why the value is expected?

For instance, when testing `planning rules` we use full explain plan for comparison rather than adding lots assertions.

Reply:
This case just follow the style of other cases in this class.
To make it simple to read, I'll add a check for the full string result in the end of case.